src/callback_to_promise.ts

```diff
@@ -1,34 +1,33 @@
-'use strict';
-
 /**
  * Given a function fn that takes a callback as its last argument, returns
  * a new version of the function that takes the callback optionally. If
  * the function is not called with a callback for the last argument, the
  * function will return a promise instead.
  */
-function callbackToPromise(fn, context, callbackArgIndex) {
+function callbackToPromise(fn, context, callbackArgIndex = void 0): any {
```

Comment:
So `callbackArgIndex = void 0` here is for typescript function overloading. Another way I could write this is `callbackArgIndex?` to mark it optional which is what I meant.

Since so many calls in airtable.js to this function do not include the third argument I wanted to mark that in its type somehow. I opted for the explicit `undefined` default since the if tests with void 0 explicitly. I figured that might be less confusing for someone to read.

-----

I really like your Symbol suggestion if we want to make to execution changes in a follow up to this PR.